pkg/provisioningv2/rke2/configserver/custom.go

```diff
@@ -35,13 +35,7 @@ func (r *RKE2ConfigServer) findMachineByClusterToken(req *http.Request) (string,
 		return "", "", err
 	}
 
-	data := map[string]interface{}{}
-	for k, v := range req.Header {
-		if strings.HasPrefix(k, headerPrefix) {
-			data[strings.ToLower(strings.TrimPrefix(k, headerPrefix))] = v
-		}
-	}
-	data["id"] = machineID
```

Comment:
still need to set the machineid?

Reply:
It was already being set above: the header is `X-Cattle-Id` which is handled by the `TrimPrefix` and `ToLower` calls.

Reply:
Gtg then